File: app/web/src/store/member.store.ts
import { create } from 'zustand';
import { memberService } from '../services/member.service';
import type { Member, BalanceRecord, ScoreRecord } from '../types';

interface MemberState {
  profile: Member | null;
  balanceRecords: BalanceRecord[];
  scoreRecords: ScoreRecord[];
  isLoading: boolean;
  
  // Actions
  fetchProfile: () => Promise<void>;
  updateProfile: (data: Partial<Member>) => Promise<void>;
  fetchBalanceRecords: () => Promise<void>;
  fetchScoreRecords: () => Promise<void>;
}

export const useMemberStore = create<MemberState>((set) => ({
  profile: null,
  balanceRecords: [],
  scoreRecords: [],
  isLoading: false,

  // 获取会员信息
  fetchProfile: async () => {
    set({ isLoading: true });
    try {
      const response = await memberService.getProfile();
      set({ profile: response.data, isLoading: false });
    } catch (error) {
      set({ isLoading: false });
      throw error;
    }
  },

  // 更新会员信息
  updateProfile: async (data: Partial<Member>) => {
    set({ isLoading: true });
    try {
      const response = await memberService.updateProfile(data);
      set({ profile: response.data, isLoading: false });
    } catch (error) {
      set({ isLoading: false });
      throw error;
    }
  },

  // 获取余额记录
  fetchBalanceRecords: async () => {
    set({ isLoading: true });
    try {
      const response = await memberService.getBalanceRecords();
      set({ balanceRecords: response.data || [], isLoading: false });
    } catch (error) {
      set({ isLoading: false });
      throw error;
    }
  },

  // 获取积分记录
  fetchScoreRecords: async () => {
    set({ isLoading: true });
    try {
      const response = await memberService.getScoreRecords();
      set({ scoreRecords: response.data || [], isLoading: false });
    } catch (error) {
      set({ isLoading: false });
      throw error;
    }
  },
}));

File: app/ios/ZayumMember/ZayumMember/Store/MemberStore.swift
//
//  MemberStore.swift
//  ZayumMember
//
//  Created by Kimi on 2026/3/27.
//

import Foundation
import Observation

@Observable
class MemberStore {
    static let shared = MemberStore()
    
    var profile: Member?
    var balanceRecords: [BalanceRecord] = []
    var scoreRecords: [ScoreRecord] = []
    var isLoading: Bool = false
    var errorMessage: String?
    
    private let memberService = MemberService.shared
    
    private init() {}
    
    // MARK: - 获取用户资料
    func fetchProfile() async {
        isLoading = true
        errorMessage = nil
        do {
            let member = try await memberService.getProfile()
            profile = member
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }
    
    // MARK: - 更新用户资料
    func updateProfile(nickname: String?, email: String?, gender: String?, birthday: String?) async {
        isLoading = true
        errorMessage = nil
        
        // 转换 gender 字符串为 Gender 枚举
        let genderEnum: Gender? = {
            guard let gender = gender else { return nil }
            switch gender {
            case "1": return .male
            case "2": return .female
            default: return .unknown
            }
        }()
        
        do {
            let updatedProfile = try await memberService.updateProfile(
                nickname: nickname,
                email: email,
                gender: genderEnum,
                birthday: birthday
            )
            profile = updatedProfile
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }
    
    // MARK: - 获取余额记录
    func fetchBalanceRecords() async {
        isLoading = true
        errorMessage = nil
        do {
            let records = try await memberService.getBalanceRecords()
            balanceRecords = records
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }
    
    // MARK: - 获取积分记录
    func fetchScoreRecords() async {
        isLoading = true
        errorMessage = nil
        do {
            let records = try await memberService.getScoreRecords()
            scoreRecords = records
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }
}
